onnxruntime/core/providers/nnapi/nnapi_builtin/builders/qdq_support_helper.cc

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include <core/graph/graph_viewer.h>
+#include <core/providers/common.h>
+
+#include "qdq_support_helper.h"
+
+namespace onnxruntime {
+namespace nnapi {
+
+using std::string;
+using std::vector;
+
+using OpVersionsMap = std::unordered_map<std::string, std::vector<ONNX_NAMESPACE::OperatorSetVersion>>;
+
+QDQSupportHelper::QDQSupportHelper(Selectors&& selectors)
+    : selectors_{std::move(selectors)} {
+  for (const auto& entry : selectors_.SelectorsSet()) {
+    for (const auto& op_info : entry->op_versions_map) {
+      bool inserted = op_type_to_selectors_map_.insert({op_info.first, &*entry}).second;
+
+      ORT_ENFORCE(inserted, "Multiple entries for operator is not supported. OpType=", op_info.first);
+    }
+  }
+}
+
+void Selectors::RegisterSelector(const Selector::OpVersionsMap& ops_and_versions_in,
+                                 std::unique_ptr<QDQ::BaseSelector> selector_in) {
+  auto entry = std::make_unique<Selector>(
+      ops_and_versions_in,
+      std::move(selector_in));
+
+  ORT_IGNORE_RETURN_VALUE(selectors_set_.insert(std::move(entry)));
+}
+
+QDQ::NodeGroup QDQSupportHelper::Match(const GraphViewer& graph_viewer, const Node& node) const {
```

Comment:
>Match [](http://example.com/codeflow?start=33&length=5)

this returns an empty NodeGroup if unmatched. then GetQDQNodeGroup will process that into an empty NodeGroupNonIndex and insert that into target_node_to_qdq_group_.

how about indicating the non-matched case in the result here? e.g., `optional<NodeGroup>`. and maybe target_node_to_qdq_group_ doesn't need to contain empty values